ui/src/message_popup/renderers/apples_text_response.jsx

```diff
@@ -0,0 +1,48 @@
+/* @flow weak */
+import React from 'react';
+
+import MinimalTextResponse from './minimal_text_response.jsx';
+
+
+// This double-logs text responses for Apples-to-Apples style group reviewing.
+// It's intended to be have the same API as MinimalTextResponse.
+export default React.createClass({
+  displayName: 'ApplesTextResponse',
+
+  propTypes: {
+    applesSceneNumber: React.PropTypes.number.isRequired,
+    onLogMessage: React.PropTypes.func.isRequired,
+    onResponseSubmitted: React.PropTypes.func.isRequired,
+    forceResponse: React.PropTypes.bool,
+    responsePrompt: React.PropTypes.string,
+    recordText: React.PropTypes.string,
+    ignoreText: React.PropTypes.string,
+    autoFocus: React.PropTypes.bool,
+    textHeight: React.PropTypes.number
+  },
+
+
+  // Watch for text responses, and double-log them.
+  onApplesLogMessage(type, response) {
```

Comment:
This wraps the `MinimalTextResponse` component, listens for some data it broadcasts out, and then adds in some other data that's needed about the scene number or whether the user clicked "Move on" so we can show that in the review at the end as well.